### ISSUE TITLE
false positive redundant_field_names when using derive-new

### ISSUE THREAD

**Issue by topecongiro** [rust-lang-nursery/rust-clippy#2516]
If you run `cargo clippy` against the following code,

```rust
#[macro_use]
extern crate derive_new;

#[derive(new)]
pub struct Foo {
    bar: String,
}
```

you will get the following error warning:

```
warning: redundant field names in struct initialization
  --> src/lib.rs:10:10
   |
10 | #[derive(new)]
   |          ^^^ help: replace it with: `bar`
   |
   = note: #[warn(redundant_field_names)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#redundant_field_names
```

Using `0.0.187`.

**Comment by oli-obk** [comment]
needs a macro check
